wgpu-core/src/instance.rs

```diff
@@ -67,8 +67,12 @@ pub struct Instance {
 }
 
 impl Instance {
-    pub fn new(name: &str, backends: Backends) -> Self {
-        fn init<A: HalApi>(_: A, mask: Backends) -> Option<A::Instance> {
+    pub fn new(name: &str, instance_options: wgt::InstanceOptions) -> Self {
+        fn init<A: HalApi>(
+            _: A,
+            mask: Backends,
+            dx12_shader_compiler: Option<wgt::Dx12Compiler>,
```

Comment:
I feel like it's cleaner to pass `None` to all the non dx12 backends here, rather than passing it in and just having them ignore it (though I guess they still have the field on their `InstanceDescriptor`).

Reply:
So it might make sense to pass in `InstanceOptions` here if the `InstanceOptions` struct is going to get bigger, to avoid the `init()` call here from getting too verbose, but you'll still have to destructure it in the `init()` function here to transform it into a `hal::InstanceDescriptor`. IMO passing in None for backends where the feature being passed in isn't used is also clearer, but I guess it doesn't matter that much.

Reply:
This passes in `wgt::InstanceDescriptor` now, but changes `hal::InstanceDescriptor` to use `Dx12Compiler` instead of `Option<Dx12Compiler>`.